src/Akka.Persistence.Sql/Query/QueryThrottler.cs

```diff
@@ -95,19 +173,29 @@ private void QueryStartGranted(IActorRef actorRef)
         actorRef.Tell(Query.QueryStartGranted.Instance);
     }
     
-    private void ReturnQueryPermit(IActorRef actorRef)
+    private void ReturnQueryPermit()
     {
         _usedPermits--;
-        Context.Unwatch(actorRef);
 
         if (_usedPermits < 0)
-            throw new IllegalStateException("Permits must not be negative");
+        {
+            _log.Warning("Permits must not be negative");
+            _usedPermits = 0;
+            return;
+        }
 
-        var popRef = _pending.First?.Value;
-        if (popRef is not null)
+        while (_pending.First is not null)
         {
+            var pending = _pending.First.Value;
             _pending.RemoveFirst();
-            QueryStartGranted(popRef);
+            if (pending is not null && pending.IsExpired)
```

Comment:
Since we can't rely on death watch mechanism to automatically purge pending requests, we have to add timeout deadlines to the pending requests. All expired pending requests will be discarded.